utils/grpc/setup_server.go

```diff
@@ -64,13 +66,29 @@ func (s *GrpcServer) GrpcServer(port string, dbURLs map[string]string, rdURL str
 
 	s.lis = lis
 
-	s.CreateServer(s.serverName, options...).
-		CreateHealthServer().
-		RegisterServer(server).
-		RegisterHealthServer().
-		WithReflection().
-		WithGracefulTermDetectAndExec().
-		AcceptConnection()
+	done := make(chan bool)
+
+	go func() {
+		s.CreateServer(s.serverName, options...).
+			CreateHealthServer().
+			RegisterServer(server).
+			RegisterHealthServer().
+			WithReflection().
+			WithGracefulTermDetectAndExec().
+			AcceptConnection()
+		done <- true
+	}()
+
+	ctx := context.Background()
+	req := new(shared.SignedRuntimeInfoRequest)
+	connection := db.ConnectionUrls{RdURL: rdURL, PgURL: dbURLs}
+
+	_, err = utils.CheckDBConnection(ctx, req, connection)
+	if err != nil {
+		log.Panic("Got error", zap.Error(err))
+	}
+
+	<-done
```

Comment:
to be safe, suggest using this and a reasonable setup timeout

```go
select {
    case <-ctx.Done():
        xxxxxx
    case <-done:
}

Reply:
@taiyangc we create a grpc server on a separate goroutine. We listen on `done` so that our main thread does not exit out. Hence, done is called by goroutine after grpcServer is terminated.
As grpcServer is a long living process, wont timeout kill the server prematurely.

Reply:
ah ok, thx for the explanation.